frontend/src/components/EmailInput.tsx

```diff
@@ -0,0 +1,59 @@
+import React, { useState, useCallback } from 'react';
+import debounce from 'lodash.debounce';
+
+interface EmailInputProps {
+  value: string;
+  onChange: (value: string) => void;
+}
+
+interface CheckEmailResponse {
+  exists: boolean;
+}
+
+const EmailInput: React.FC<EmailInputProps> = ({ value, onChange }) => {
+  const [status, setStatus] = useState<'checking' | 'available' | 'exists' | null>(null);
+
+  const checkEmail = useCallback(
+    debounce(async (email: string) => {
+      if (!email) return setStatus(null);
+      setStatus('checking');
+      try {
+      await new Promise((r) => setTimeout(r, 800)); 
+      const exists = email.includes("test") || email.endsWith("@gmail.com"); 
```

Comment:
Existence is currently derived from email.includes("test") || email.endsWith("@gmail.com") - a hard-coded rule that is neither documented as a mock nor safe for production (false positives e.g., many real users use gmail).

Mark clearly as a mock and add a TODO to replace with API call.